src/StepperFirmata2.cpp

```diff
@@ -0,0 +1,401 @@
+    
+    Firmata.write(START_SYSEX);
+    Firmata.write(ACCELSTEPPER_DATA);
+    if (complete) {
+      Firmata.write(STEPPER_MOVE_COMPLETE);
+    } else {
+      Firmata.write(STEPPER_REPORT_POSITION);
+    }
+    Firmata.write(deviceNum);
+    Firmata.write(data[0]);
+    Firmata.write(data[1]);
+    Firmata.write(data[2]);
+    Firmata.write(data[3]);
+    Firmata.write(data[4]);
+    Firmata.write(END_SYSEX);
+  }
+}
+
+void StepperFirmata2::reportGroupComplete(byte deviceNum)
+{
+  if (group[deviceNum]) {
+    
+    Firmata.write(START_SYSEX);
+    Firmata.write(ACCELSTEPPER_DATA);
+    Firmata.write(MULTISTEPPER_MOVE_COMPLETE);
+    Firmata.write(deviceNum);
+    Firmata.write(END_SYSEX);
+  }
+}
+
+/*==============================================================================
+ * SYSEX-BASED commands
+ *============================================================================*/
+
+boolean StepperFirmata2::handleSysex(byte command, byte argc, byte *argv)
+{
+    
+  if (command == ACCELSTEPPER_DATA) {
+
+    byte stepCommand, deviceNum, directionPin, stepPin, stepDirection, groupNum;
+    byte interface, wireCount, stepType;
+    byte motorPin3, motorPin4, enablePin, invertPins = 0;
+    unsigned int minPulseWidth;
+    long numSteps;
+    int maxSpeed, stepSpeed, accel, decel;
+
+    unsigned int index = 0;
+
+    stepCommand = argv[index++];
+    deviceNum = argv[index++];
+  
+    if (deviceNum < MAX_STEPPERS) {
+
+      if (stepCommand == STEPPER_CONFIG) {
+
+        interface = argv[index++]; // upper 3 bits are the wire count, bits 4-6 are the step type, lower bit indicates enable pin
+        wireCount = (interface & 0x70) >> 4;
+        stepType = interface & 0x0E;
+        directionPin = argv[index++]; // Direction or motorPin1 for TWO_WIRE or FOUR_WIRE interface
+        stepPin = argv[index++]; // Step or motorPin2 for TWO_WIRE or FOUR_WIRE interface
+
+        if (Firmata.getPinMode(directionPin) == PIN_MODE_IGNORE || Firmata.getPinMode(stepPin) == PIN_MODE_IGNORE)
+          return false;
+
+        Firmata.setPinMode(directionPin, PIN_MODE_STEPPER);
+        Firmata.setPinMode(stepPin, PIN_MODE_STEPPER);
+
+        if (!stepper[deviceNum]) {
+          numSteppers++;
+        }
+
+        if (wireCount >= 3) {
+          motorPin3 = argv[index++]; // byte 10
+          if (Firmata.getPinMode(motorPin3) == PIN_MODE_IGNORE)
+            return false;
+          Firmata.setPinMode(motorPin3, PIN_MODE_STEPPER);
+        }
+
+        if (wireCount >= 4) {
+          motorPin4 = argv[index++]; // byte 11
+          if (Firmata.getPinMode(motorPin4) == PIN_MODE_IGNORE)
+            return false;
+          Firmata.setPinMode(motorPin4, PIN_MODE_STEPPER);
+        }
+
+        if (interface & 0x01) {
+          enablePin = argv[index++]; // byte 10, 11 or 12 iff interface bit 7 is 1
+        }
+
+        if (wireCount == 0) {
+          stepper[deviceNum] = new AccelStepper(AccelStepper::DRIVER, directionPin, stepPin);
```

Comment:
Rather than just swap the parameters here I made the order consistent in protocol as well. It bothered me having a signature that looked like
````
stepper[deviceNum] = new AccelStepper(AccelStepper::DRIVER, directionOrMotorPin2, stepOrMotorPin1);
````